modules/dynamic/src/smithy4s/dynamic/internals/DynamicService.scala

```diff
@@ -23,25 +23,19 @@ import smithy4s.kinds.PolyFunction5
 private[internals] case class DynamicService(
     id: ShapeId,
     version: String,
-    endpoints: List[DynamicEndpoint],
+    endpoints: Vector[DynamicEndpoint],
     hints: Hints
 ) extends Service.Reflective[DynamicOp]
     with DynamicSchemaIndex.ServiceWrapper {
 
   type Alg[P[_, _, _, _, _]] = PolyFunction5.From[DynamicOp]#Algebra[P]
   override val service: Service[Alg] = this
 
-  private lazy val endpointMap: Map[ShapeId, Endpoint[_, _, _, _, _]] =
-    endpoints.map(ep => ep.id -> ep).toMap
+  private lazy val ordinalMap: Map[ShapeId, Int] =
```

Comment:
this could be changed by finding a way for the `DynamicEndpoint` to carry the ordinal

Reply:
is there any issue with making it do so?

Reply:
although tbh I don't mind it either way, the map seems just as fine

Reply:
> is there any issue with making it do so? 

It's not that simple, because the DynamicOps are not currently tied to a service.